src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -711,17 +707,15 @@ func (s *fileSystemSource) readNextEntryAndMaybeIndex(
 	r fs.DataFileSetReader,
 	batch []doc.Document,
 	builder *result.IndexBuilder,
+	tagDecoder serialize.TagDecoder,
 ) ([]doc.Document, error) {
 	// If performing index run, then simply read the metadata and add to segment.
-	id, tagsIter, _, _, err := r.ReadMetadata()
+	id, encodedTags, _, _, err := r.StreamingReadMetadata()
 	if err != nil {
 		return batch, err
 	}
 
-	d, err := convert.FromSeriesIDAndTagIter(id, tagsIter)
-	// Finalize the ID and tags.
-	id.Finalize()
-	tagsIter.Close()
```

Comment:
Should the id and encoded tags be finalized here still, considering `FromRawSeriesIDAndTags` is cloning them?

Reply:
With streaming reads, id and tags are just raw `[]byte` slices and thus no longer a subject to finalization.